x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/conditional/Case.java

```diff
@@ -303,40 +312,58 @@ public ExpressionEvaluator get(DriverContext context) {
 
             @Override
             public String toString() {
-                return "CaseEvaluator[resultType="
-                    + resultType
-                    + ", conditions="
-                    + conditionsFactories
-                    + ", elseVal="
-                    + elseValueFactory
-                    + ']';
+                return "CaseEvaluator[conditions=" + conditionsFactories + ", elseVal=" + elseValueFactory + ']';
             }
         };
     }
 
-    record ConditionEvaluatorSupplier(ExpressionEvaluator.Factory condition, ExpressionEvaluator.Factory value)
+    record ConditionEvaluatorSupplier(Source conditionSource, ExpressionEvaluator.Factory condition, ExpressionEvaluator.Factory value)
         implements
             Function<DriverContext, ConditionEvaluator> {
         @Override
         public ConditionEvaluator apply(DriverContext driverContext) {
-            return new ConditionEvaluator(condition.get(driverContext), value.get(driverContext));
+            return new ConditionEvaluator(
+                /*
+                 * We treat failures as null just like any other failure.
+                 * It's just that we then *immediately* convert it to
+                 * true or false using the tri-valued boolean logic stuff.
+                 * And that makes it into false. This is, *exactly* what
+                 * happens in PostgreSQL and MySQL and SQLite:
+                 * > SELECT CASE WHEN null THEN 1 ELSE 2 END;
+                 * 2
+                 * Rather than go into depth about this in the warning message,
+                 * we just say "false".
```

Comment:
CC @costin and @astefan : Nik is nicely making the MV semantics for CASE consistent.

Just FYI because we could also choose to define `CASE([true, true], ... ) == null` similarly to functions, which applied to multivalues just become `null`. I think Nik's approach is more useful, because making `CASE` on multivalues always return `null` would probably also imply that `CASE(true, a, [true, true], b)` should return `null`, even though it doesn't matter that the second condition is multivalued.

Reply:
> . I think Nik's approach is more useful
It's also the same approach as PostgreSQL.

Reply:
I don't think Case makes the equivalence of null == false rather it checks if the condition is true.
That is  `SELECT CASE WHEN null THEN 1 ELSE 2 END` returns 2 because the condition `null` is not true. 
It's a subtle difference but it means `CASE(true, a, [true, true], b)` should return a because the condition is `true` so the else branch is not even evaluated.